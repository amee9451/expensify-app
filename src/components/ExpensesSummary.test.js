import React from 'react';
import { shallow } from 'enzyme';

import { ExpensesSummary } from './ExpensesSummary';

let wrapper;

beforeEach(() => {
  wrapper = shallow(
    <ExpensesSummary expensesCount={0} expensesTotal={0} />
  );
});

test('should render ExpensesSummary correctly with no expenses', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary correctly with one expense', () => {
  wrapper.setProps({
    expensesCount: 1,
    expensesTotal: 9200
  });

  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary correctly with multiple expenses', () => {
  wrapper.setProps({
    expensesCount: 3,
    expensesTotal: 1200
  });

  expect(wrapper).toMatchSnapshot();
});