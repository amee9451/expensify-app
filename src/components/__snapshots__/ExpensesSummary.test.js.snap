// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <strong>
        3
      </strong>
       
      expenses
       totalling 
      <strong>
        $12.00
      </strong>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render ExpensesSummary correctly with no expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <strong>
        0
      </strong>
       
      expenses
       totalling 
      <strong>
        $0.00
      </strong>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render ExpensesSummary correctly with one expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <strong>
        1
      </strong>
       
      expense
       totalling 
      <strong>
        $92.00
      </strong>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
